platform/web/js/libs/library_godot_audio.js

```diff
@@ -453,6 +462,36 @@ class SampleNode {
 		const bus = GodotAudio.Bus.getBus(params.busIndex);
 		const sampleNodeBus = this.getSampleNodeBus(bus);
 		sampleNodeBus.setVolume(options.volume);
+
+		try {
+			this._positionWorklet = this.createPositionWorklet();
+			this.finalize(options);
+		} catch (e) {
+			const path = GodotConfig.locate_file('godot.audio.position.worklet.js');
+			GodotAudio.ctx.audioWorklet
+				.addModule(path)
+				.then(() => {
+					if (!this.isCanceled) {
+						this._positionWorklet = this.createPositionWorklet();
+						this.finalize(options);
+					}
+				}).catch((e2) => {
+					GodotRuntime.error('Failed to create Position Worklet.', e2);
+				});
+		}
+	}
+
+	/**
+	 * Finalizes Node setup and connects PositionWorklet node
+	 */
+	finalize(options) {
+		if (this._positionWorklet) {
+			this._source.connect(this._positionWorklet);
+			if (options.start) {
+				this.start();
+				this.isStarted = true;
+			}
+		}
```

Comment:
Fair, was just trying to avoid duplicating the code like this, but i guess it not much:

```js
try {
			this._positionWorklet = this.createPositionWorklet();
			this._source.connect(this._positionWorklet);
			if (options.start) {
				this.start();
			}			
		} catch (e) {
			const path = GodotConfig.locate_file('godot.audio.position.worklet.js');
			GodotAudio.ctx.audioWorklet
				.addModule(path)
				.then(() => {
					if (!this.isCanceled) {
						this._positionWorklet = this.createPositionWorklet();
						this._source.connect(this._positionWorklet);
						if (options.start) {
							this.start();
						}
					}
				}).catch((e2) => {
					GodotRuntime.error('Failed to create Position Worklet.', e2);
				});
		}
```